src/controllers/hid/hidcontroller.h

```diff
@@ -60,21 +57,58 @@ class HidControllerJSProxy : public ControllerJSProxy {
               m_pHidController(m_pController) {
     }
 
-    Q_INVOKABLE void send(const QList<int>& data, unsigned int length = 0) override {
-        m_pHidController->send(data, length);
+    /// @brief Sends HID OutputReport with hard coded ReportID 0 to HID device
+    ///        This function only works with HID devices, which don't use ReportIDs
+    /// @param dataList Data to send as list of bytes
+    /// @param length Unused optional argument
+    Q_INVOKABLE void send(const QList<int>& dataList, unsigned int length = 0) override {
+        // This function is only for class compatibility with the (midi)controller
+        Q_UNUSED(length);
+        this->send(dataList, 0, 0);
+    }
+
+    /// @brief Sends HID OutputReport to HID device
+    /// @param dataList Data to send as list of bytes
+    /// @param length Unused but mandatory argument
+    /// @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+    /// @param skipIdenticalReports If set, the sending of the report is inhibited, when the data didn't changed
+    Q_INVOKABLE void send(const QList<int>& dataList,
+            unsigned int length,
+            unsigned int reportID,
+            bool skipIdenticalReports = true) {
+        Q_UNUSED(length);
+        QByteArray dataArray;
+        dataArray.reserve(dataList.size());
+        for (int datum : dataList) {
+            dataArray.append(datum);
+        }
+        this->sendOutputReport(dataArray, reportID, skipIdenticalReports);
     }
 
-    Q_INVOKABLE void send(const QList<int>& data, unsigned int length, unsigned int reportID) {
-        m_pHidController->sendReport(data, length, reportID);
+    /// @brief Sends an OutputReport to HID device
+    /// @param dataArray Data to send as byte array (Javascript type Uint8Array)
+    /// @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+    /// @param skipIdenticalReports If set, the sending of the report is inhibited, when the data didn't changed
+    Q_INVOKABLE void sendOutputReport(const QByteArray& dataArray,
+            unsigned int reportID,
+            bool skipIdenticalReports = true) {
```

Comment:
`resendUnchangedReport = false`

- "identity" is a different concept
- using `false` as the default value is more common
- `true` triggers an action -> positive logic
- only applies to a single report, not multiple